crates/ruff_python_ast/src/visitor/preorder.rs

```diff
@@ -0,0 +1,1101 @@
+use crate::prelude::*;
+
+/// Visitor that traverses all nodes recursively in pre-order.
+pub trait PreorderVisitor<'a> {
+    fn visit_mod(&mut self, module: &'a Mod) {
+        walk_module(self, module);
+    }
+
+    fn visit_stmt(&mut self, stmt: &'a Stmt) {
+        walk_stmt(self, stmt);
+    }
+
```

Comment:
(Inconsistent spacing between methods :))

Reply:
Lol yeah. I copied the `Visitor` methods because I'm lazy but I prefer to have empty lines so I started adding them. It's a pitty rustfmt doesn't add them automatically, now I (the lazy person) has to do it.